src/data/apigateway/mod.rs

```diff
@@ -0,0 +1,170 @@
+#[cfg(test)]
+mod tests;
+
+pub mod auth;
+
+use chrono::prelude::*;
+use std::fmt;
+use std::collections::HashMap;
+use serde;
+use serde_qs as qs;
+
+#[derive(Debug,Eq,PartialEq,Serialize,Deserialize)]
+pub enum HttpMethod {
+    HEAD,
+    GET,
+    POST,
+    PUT,
+    OPTIONS,
+    DELETE,
+}
+
+impl fmt::Display for HttpMethod {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", match *self {
+            HttpMethod::HEAD => "HEAD",
+            HttpMethod::GET => "GET",
+            HttpMethod::POST => "POST",
+            HttpMethod::PUT => "PUT",
+            HttpMethod::OPTIONS => "OPTIONS",
+            HttpMethod::DELETE => "DELETE"
+        })
+    }
+}
+
+#[derive(Serialize,Deserialize)]
+pub struct HttpEvent {
+    pub body: Option<String>,
+    pub headers: Option<HashMap<String, String>>,
```

Comment:
One small thought. I did something similar initially in a separate project and found that the extra option type to be a bit annoying from a usability perspective. Espectially those new to rust

```
9 |     println!("{:?}", ev.query_string_parameters.and_then(|m| m.get("foo")));
  |     ---------------------------------------------------------^--------------
  |     |                                                        |          |
  |     |                                                        |          `m` dropped here while still borrowed
  |     |                                                        borrowed value does not live long enough
  |     borrowed value needs to live until here
```

Was't a very great experience.

I found a way to [deserialize the nullable maps into default values](https://github.com/softprops/lando/blob/master/src/request.rs#L15-L16) greatly improved usability for new comers

as well as reduced verbosity while retaining the same level of rust runtime saftey.

```rust
 println!("{:?}", ev.query_string_parameters.and_then(|m| m.get("foo")))
```

```rust
println!("{:?}", ev.query_string_parameters.get("foo"));
```

Reply:
Yes, let's deserialize things like maps and vectors into default (empty) values.

Reply:
I really think that kind of thing helps the ergonomics for people using this.